examples/chip-tool/commands/clusters/ReportCommand.h

```diff
@@ -98,6 +98,8 @@ class ReportCommand : public InteractionModelReports, public ModelCommand, publi
 
     void OnError(CHIP_ERROR error) override
     {
+        ReturnOnFailure(RemoteDataModelLogger::LogErrorAsJSON(error));
```

Comment:
`ReturnErrorOnFailure` is supposed to return the error ... but this method returns void.

Reply:
Why does compilation pass on this? I would have expected this to not compile since `return __err` on a function returning void should not be allowed. Am I reading things wrong?

Reply:
Mis-read ... `ReturnOnFailure` looks very close with `ReturnErrorOnFailure'